### PR TITLE
RATEST-66: Write attachments feature gherkin scenarios

### PR DESCRIPTION
**Pull request by kdaud**
Here is the ticket id  https://issues.openmrs.org/browse/RATEST-66

### CODE REVIEW

File: src/test/resources/features/15-attachment/attachements.feature [hunk 1]
```diff
@@ -0,0 +1,17 @@
+ Feature: Attachments Management
+
+   Background:
+     Given a user logs in, searches a patient and visits attachment page 
+     When a user clicks on attachment menu
+     Then the system loads the page
```
Comment (kaweesi) [inline src/test/resources/features/15-attachment/attachements.feature:6]: you can be more specific of which page you mean here

File: src/test/resources/features/15-attachment/attachements.feature [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+   Background:
+     Given a user logs in, searches a patient and visits attachment page 
+     When a user clicks on attachment menu
+     Then the system loads the page
+
+   #User performing action on the attachment page
+   Scenario: Uploading attachment
+     When a user attaches a file
+     And a user enters a caption
+     And a user clicks "Upload file" functionality
```
Comment (kaweesi) [inline src/test/resources/features/15-attachment/attachements.feature:12]: any reason for using double quotes in these statements

**Review by kaweesi — COMMENTED** [review]
Thanks @kdaud for this PR, each functionality on a page is good enough being a scenario of it's own. please catch more comments from this review

File: src/test/resources/features/15-attachment/attachements.feature (hunk 2 above)
Comment (kdaud) [reply src/test/resources/features/15-attachment/attachements.feature:12]: Thanks @k-joseph  for the review. In the nutshell am addressing the comments and soon updating the PR

File: src/test/resources/features/15-attachment/attachements.feature [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+     When a user attaches a file
+     And a user enters a caption
+     And a user clicks Upload file functionality
+     Then the system uploads the attached file
+     
+   #System clearing the forms
+   Scenario: Clearing forms
+     When a user clicks clear forms button
+     Then the system clears the forms
+     And quit browser
```
Comment (kaweesi) [inline src/test/resources/features/15-attachment/attachements.feature:19]: @kdaud cross check against existing ones to ensure your steps are all unique

Comment (kdaud) [reply src/test/resources/features/15-attachment/attachements.feature:19]: > @kdaud cross check against existing ones to ensure your steps are all unique

Hey @kaweesi I would really love to work on the recommendation but have not yet figured out its implication

**Comment by kdaud** [comment]
Hey @kaweesi have updated the PR.

File: src/test/resources/features/15-attachment/attachments.feature [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+     And a user clicks on Attachment link
+     Then the system loads Attachment page
+     And a user attaches a file
+     
+   Scenario: Uploading Attachment
+     And a user enters a caption
+     Then the system enables upload button
+     When a user clicks the upload link button 
+     Then the system uploads the attached file
+     And close browser
```
Comment (kaweesi) [inline src/test/resources/features/15-attachment/attachments.feature:14]: please remove the close step, it's taken care by the steps

File: src/test/resources/features/15-attachment/attachments.feature [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+   Background:
+     Given a user logs in, searches a patient and visits Attachment page 
+     And a user clicks on Attachment link
+     Then the system loads Attachment page
+     And a user attaches a file
+     
+   Scenario: Uploading Attachment
+     And a user enters a caption
+     Then the system enables upload button
+     When a user clicks the upload link button
```
Comment (kaweesi) [inline src/test/resources/features/15-attachment/attachments.feature:12]: can it be both a link and button?

File: src/test/resources/features/15-attachment/attachments.feature [hunk 6]
```diff
@@ -0,0 +1,19 @@
+ Feature: Attachments Management
+
+   Background:
+     Given a user logs in, searches a patient and visits Attachment page 
+     And a user clicks on Attachment link
+     Then the system loads Attachment page
+     And a user attaches a file
```
Comment (kaweesi) [inline src/test/resources/features/15-attachment/attachments.feature:7]: must every scenario require attaching a file?

File: src/test/resources/features/15-attachment/attachments.feature [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+     
+   Scenario: Uploading Attachment
+     And a user enters a caption
+     Then the system enables upload button
+     When a user clicks the upload link button 
+     Then the system uploads the attached file
+     And close browser
+     
+   Scenario: Clearing forms
```
Comment (kaweesi) [inline src/test/resources/features/15-attachment/attachments.feature:16]: try renaming this scenario to make attach more user context (meaning)

File: src/test/resources/features/15-attachment/attachments.feature (hunk 4 above)
Comment (kdaud) [reply src/test/resources/features/15-attachment/attachments.feature:14]: Have made the recommendation

File: src/test/resources/features/15-attachment/attachments.feature (hunk 5 above)
Comment (kdaud) [reply src/test/resources/features/15-attachment/attachments.feature:12]: I didn't take keen on this. However, have caught it in the next PR

File: src/test/resources/features/15-attachment/attachments.feature (hunk 6 above)
Comment (kdaud) [reply src/test/resources/features/15-attachment/attachments.feature:7]: After taking again a use case, have realised its specific on one scenario. Have refactored the code base in the PR

File: src/test/resources/features/15-attachment/attachments.feature (hunk 7 above)
Comment (kdaud) [reply src/test/resources/features/15-attachment/attachments.feature:16]: Thanks @kaweesi for reviewing my work. Have addressed the recommendation in the PR

File: src/test/resources/features/15-attachment/attachments.feature [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+     And a user enters a caption
+     Then the system enables upload button
+     When a user clicks the upload button 
+     Then the system uploads the attached file
+     
+   Scenario: Removing un intended Attachement 
+     And a user attaches a file or enters a caption
+     And a user clicks clear forms button
+     Then the system clears the form
+     And close browser
```
Comment (kaweesi) [inline src/test/resources/features/15-attachment/attachments.feature:19]: you have forgotten to get rid of the last step, we are taking care of it in the framework and through @After("selenium)

Comment (kdaud) [reply src/test/resources/features/15-attachment/attachments.feature]: Oops! I shall be more smart next time. Have refactored the codebase

File: src/test/resources/features/15-attachment/attachments.feature [hunk 9]
```diff
@@ -0,0 +1,18 @@
+ Feature: Attachments Management
+
```
Comment (kaweesi) [inline src/test/resources/features/15-attachment/attachments.feature:2]: Please add a scenario where the camera icon is used

Comment (kdaud) [reply src/test/resources/features/15-attachment/attachments.feature:2]: Am working on this and updating the pr ASP

File: src/test/resources/features/15-attachment/attachments.feature [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+     
+   Scenario: Removing un intended Attachement 
+     And a user attaches a file or enters a caption
+     And a user clicks clear forms button
+     Then the system clears the form 
+     
+   Scenario: Capturing an image
+     And a user clicks in camera icon
+     Then the system enables capturing the photo
+     And a user clicks save button
```
Comment (kaweesi) [inline src/test/resources/features/15-attachment/attachments.feature:23]: what must the user do before pressing save button

Comment (kdaud) [reply src/test/resources/features/15-attachment/attachments.feature:24]: Probably focusing in the camera

File: src/test/resources/features/15-attachment/attachments.feature [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+     When a user clicks the upload button 
+     Then the system uploads the attached file
+     
+   Scenario: Removing un intended Attachement 
+     And a user attaches a file or enters a caption
+     And a user clicks clear forms button
+     Then the system clears the form 
+     
+   Scenario: Capturing an image
+     And a user clicks in camera icon
```
Comment (kaweesi) [inline src/test/resources/features/15-attachment/attachments.feature:21]: rename this to click on

File: src/test/resources/features/15-attachment/attachments.feature [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+     Then the system enables capturing the photo
+     And a user focuses in the camera
```
Comment (kaweesi) [inline src/test/resources/features/15-attachment/attachments.feature:23]: rename from focus to something better

File: src/test/resources/features/15-attachment/attachments.feature (hunk 10 above)
Comment (kaweesi) [reply src/test/resources/features/15-attachment/attachments.feature:24]: I think am ok with just merging the efforts and have one open PR

File: src/test/resources/features/15-attachment/attachments.feature (hunk 11 above)
Comment (kdaud) [reply src/test/resources/features/15-attachment/attachments.feature:21]: Have addressed this recommendation

File: src/test/resources/features/15-attachment/attachments.feature (hunk 12 above)
Comment (kdaud) [reply src/test/resources/features/15-attachment/attachments.feature:23]: Made modification in regard to the scenario. Basing on the recommendations, I will commit the pr on this [ticket](https://issues.openmrs.org/browse/RATEST-68)

**Pull request closed without merging**